sdk/search/azure-search-documents/src/main/java/com/azure/search/documents/SearchAsyncClient.java

```diff
@@ -769,4 +779,18 @@ private static AutocompleteRequest createAutoCompleteRequest(String searchText,
             .setDocument(d)));
         return batch;
     }
+
+    private static class SearchObjectMapper {
+        private static ObjectMapper MAPPER = null;
+        private SearchObjectMapper() {
+        }
+
+        static ObjectMapper getInstance() {
+            if (MAPPER == null) {
```

Comment:
This won't be thread-safe.

Reply:
This is exactly the same setting sharing by threads. Even thread writes the MAPPER, it will not throw error but update with new one.

Reply:
Suppress spotbugs for best practice. Remove singleton and put it property